SensusUI/PromptForInputsPage.cs

```diff
@@ -83,6 +83,30 @@ public PromptForInputsPage(InputGroup inputGroup, int stepNumber, int totalSteps
 
             int viewNumber = 1;
             bool anyRequired = false;
+
+            foreach (Input input in inputGroup.Inputs)
+                if (input.Display)
+                {
+                    View inputView = input.GetView(viewNumber);
+                    if (inputView != null)
+                    {
+                        if (input.Required)
+                            anyRequired = true;
+                    }
+                }
+
+            if (_displayedInputCount > 0)
+                contentLayout.Children.Add(new BoxView { Color = Color.Transparent, HeightRequest = inputSeparatorHeight });
+
+            if (anyRequired)
```

Comment:
Reduce the above to `if (inputGroup.Inputs.Any(input => input.Display && input.Required) { ... }`.